lib/mpegts/ES_hevc.cpp

```diff
@@ -183,9 +183,15 @@ void ES_hevc::Parse_HEVC(int buf_ptr, unsigned int NumBytesInNalUnit, bool &comp
     case NAL_VPS_NUT:
       if (m_NeedVPS)
       {
-        memcpy(stream_info.extra_data + stream_info.extra_data_size, es_buf + (buf_ptr - 4), NumBytesInNalUnit);
-        stream_info.extra_data_size += NumBytesInNalUnit;
-        m_NeedVPS = false;
+        if (stream_info.extra_data_size + NumBytesInNalUnit <= sizeof(stream_info.extra_data))
+        {
+          memcpy(stream_info.extra_data + stream_info.extra_data_size, es_buf + (buf_ptr - 4), NumBytesInNalUnit);
+          stream_info.extra_data_size += NumBytesInNalUnit;
+          m_NeedVPS = false;
+        }
+        else {
```

Comment:
{ on a new line please, same for the ones below.

Reply:
Done. I also did a quick scan in the project, you might want to have a look in src/helpers.cpp